### PR TITLE
Issue 252: Deprecate test-helper module in favour of test-fixture-maker module

### PR DESCRIPTION
**Pull request by OldSneerJaw**
Introduced the test-fixture-maker module, which is superior to the test-helper module; it produces isolated test fixtures so that state is not accidentally shared between test suites and test cases. The test-helper module has been deprecated.

Fully addresses issue #252.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -35,7 +35,7 @@ To learn more about Sync Gateway, check out [Couchbase](http://www.couchbase.com
 
 Synctos is distributed as an [npm](https://www.npmjs.com/) package and it requires that [Node.js](https://nodejs.org/) is installed in order to run. NOTE: The minimum officially supported version of Node.js is v8.9.0.
 
-To add synctos to your project, run `npm install synctos` from the project's root directory to install the package locally. Or, better yet, if you define a package.json file in your project, you can run `npm install synctos --savedev` to automatically install locally and insert the package into your package.json's developer dependencies.
+To add synctos to your project, run `npm install synctos` from the project's root directory to install the package locally. Or, better yet, if you define a `package.json` file in your project, you can run `npm install synctos --savedev` to automatically install locally and insert the package into your `package.json` file's developer dependencies.
```
[the remark this answers is not in the document]
Comment (OldSneerJaw) [reply README.md]: Right you are! I'll fix that.

**Comment by OldSneerJaw** [comment]
I have examined the time zone issue in greater detail and determined that the current behaviour is not a bug. It boils down to this: when an ISO 8601 date-time string has a time component but no time zone component, the ECMAScript specification is vague on the details of which time zone should be applied in such situations. The options seem to be either the server's **current** local time zone or the server's local time zone **as it would be/would have been on the date/time represented by the string value**, which may differ due to Daylight Saving Time. Most JavaScript implementations seem to prefer the latter approach but, as I noted, it does not seem to be required by the specification. For that reason and because this is an edge case (a client _should_ always specify a time zone since they have no control over which time zone the server is in), I have opted for the simpler approach and instead use the server's current local time zone in such cases.

TL;DR: Ready for review again, @dkichler.

**Comment by dkichler** [comment]
And to think, this is exactly the sort of thing standards should be designed to overcome 😒  I'm particularly sympathetic in this case b/c we are dealing with similar issues with ISO 8601 on a couple fronts.
